pyqtorch/composite/compose.py

```diff
@@ -215,6 +216,109 @@ def tensor(
+                            pauli_strings[ind_op][
+                                qubit_support.index(subop.qubit_support[0])
+                            ]
+                            == ""
+                        ):
+                            pauli_strings[ind_op][
+                                qubit_support.index(subop.qubit_support[0])
+                            ] = subop.name
+                        else:
+                            # cases where two or more operators
+                            # are defined for a same qubit
+                            # are not considered
+                            return False
+                else:
+                    if (
+                        pauli_strings[ind_op][qubit_support.index(op.qubit_support[0])]
+                        == ""
+                    ):
+                        pauli_strings[ind_op][
+                            qubit_support.index(op.qubit_support[0])
+                        ] = op.name
+                    else:
+                        # cases where two or more operators
+                        # are defined for a same qubit
+                        # are not considered
+                        return False
+
+            # Convert to binary vectors
+            binary_vectors = []
+            for p_string in pauli_strings:
+                x_vec = [1 if p in ["X", "Y"] else 0 for p in p_string]
+                z_vec = [1 if p in ["Z", "Y"] else 0 for p in p_string]
+                binary_vectors.append((x_vec, z_vec))  # Keep x and z vectors
+
+            # Check commutation using symplectic inner product
+            n_strings = len(pauli_strings)
+            for i in range(n_strings):
+                for j in range(i + 1, n_strings):
+                    vec_i = binary_vectors[i]
+                    vec_j = binary_vectors[j]
+
+                    # Split into x and z parts
+                    x_i = vec_i[0]
+                    z_i = vec_i[1]
+                    x_j = vec_j[0]
+                    z_j = vec_j[1]
+
+                    # Symplectic product: (x_i · z_j) XOR (z_i · x_j)
+                    product = 0
+                    for k in range(nb_support):
+                        product ^= (x_i[k] & z_j[k]) ^ (z_i[k] & x_j[k])
+
+                    if product == 1:
+                        return False  # Found a pair that doesn't commute
+
+            return True
+
+        return False
+
+    def _disjoint_supports(self):
+        """Check if operations are defined on disjoint supports."""
+        qubit_supports_intersect = [set(op.qubit_support) for op in self.operations]
+
+        disjoint_sets: set[int] = set()
+        for s in qubit_supports_intersect:
+            if s.isdisjoint(disjoint_sets):
+                disjoint_sets.update(s)
+            else:
+                return False
+        if len(disjoint_sets) == len(self.qubit_support):
+            return True
+
+        return False
+
+    @property
```

Comment:
Not too sure this should be a `@property`. If this is a regular predicate method, it conveys the idea that you compute smtg, which you actually do here.